### ISSUE TITLE
[Rules] Flag issue when redundant role used

### ISSUE THREAD

**Issue by tombrunet** [IBMa/equal-access#596]
New Rule: Redundancy doesn't cause failures since it will show up the same in the accessibility hierarchy. However, it's redundant and could cause some problems later. Should flag as a recommendation (maybe needs review?).

So, when getImplicitRole returns the same value as the explicit role, we should trigger a rule.

For example:
```
<button role="button">My Button</button>
```
`role="button"` isn't an error per se, but unnecessary.

**Comment by tombrunet** [comment]
Should be a Recommendation, not a Violation

**Comment by aliunwala** [comment]
In the Triage meeting:
* We think this should flagged as a recommendation

**Comment by shunguoy** [comment]
this is covered in #789
